packages/gestalt/src/ButtonGroup.js

```diff
@@ -0,0 +1,41 @@
+// @flow strict
+
+import React, { Children, type Node } from 'react';
+import PropTypes from 'prop-types';
+import Box from './Box.js';
+
+function ButtonGroup({ children }: {| children?: Node |}): Node {
+  const count = Children.count(children);
+
+  if (count === 0) {
+    return null;
+  }
+  if (count === 1) {
+    return <>{children}</>;
+  }
+
+  return (
+    <Box
+      marginStart={-1}
+      marginRight={-1}
+      marginTop={-1}
+      marginBottom={-1}
+      display="flex"
+      wrap
+    >
+      {Children.map(children, child =>
```

Comment:
Should we check the children are Buttons? 
![image](https://user-images.githubusercontent.com/10593890/92178142-c0c38c00-edf6-11ea-8c70-7eebd33d1b8b.png)

Reply:
@AlbertCarreras when we tried that in the past, it always ended up being to restrictive (e.g checking with flow / doing a typeof):
* When people want to experiment with a new button in the Pinterest codebase.
* When someone wants to wrap the button with a flyout: `<Flyout><Button / ></Flyout`
* When you wrap it in a conditional, depending on an experiment `{inExperiment ? <Button /> : null}`

Reply:
Makes sense.